htdocs/editors/CKeditor/formCkeditor.php

```diff
@@ -112,8 +112,54 @@ public function render() {
 
 		$ret = $xoTheme->addScript("/editors/CKeditor/ckeditor/ckeditor.js", array('type' => 'text/javascript'), '');
 		$ret .= $xoTheme->addScript("/editors/CKeditor/ckeditor/adapters/jquery.js", array('type' => 'text/javascript'), '');
-		$ret .= $xoTheme->addScript('', array('type' => 'text/javascript'),
-			'var config = {filebrowserImageBrowseUrl: "' . ICMS_URL . '/editors/CKeditor/imagebrowser.php", toolbar: "' .  $toolbar . '"}; $(function() { $("#'.@$this->_name.'_tarea").ckeditor(config); $("#'.@$this->_name.'_tarea").parents("form").submit(function() { var data = $("#'.@$this->_name.'_tarea").html(); $("#'.@$this->_name.'_tarea").html(data); }); });');
+
+		// CSRF token for AJAX: createToken() already returns the hashed token expected by validateToken()
+		$__tokenVal = icms::$security->createToken(0);
+
+		// Build JS with real newlines to avoid invalid escape sequences in the browser.
+		$textareaId = $this->_name . '_tarea';
+		$icmsUrl = ICMS_URL;
+		$script = <<<JS
+var config = {
+  filebrowserImageBrowseUrl: "{$icmsUrl}/editors/CKeditor/imagebrowser.php",
+  toolbar: "{$toolbar}",
+  extraPlugins: "mentions,autocomplete,textmatch"
+};
+config.mentions = [{
+  marker: "@",
+  minChars: 1,
+  throttle: 200,
+  itemTemplate: "<li data-id=\"{id}\"><img src=\"{avatar}\" style=\"width:20px;height:20px;border-radius:50%;vertical-align:middle;margin-right:6px;\" alt=\"{name}\"> @{username} <small style=\"opacity:.7;\">{name}</small></li>",
+  outputTemplate: "<a href=\"{$icmsUrl}/userinfo.php?uid={id}\" class=\"mention\">@{username}</a>",
+  feed: function(opts, callback) {
+    var xhr = new XMLHttpRequest();
+    // The 'autocomplete' instance is available via `this` in the dataCallback context.
+    var requestId = this.autocomplete.model.lastRequestId;
+    var url = "{$icmsUrl}/include/ajax/mentions.php?q=" + encodeURIComponent(opts.query) + "&XOOPS_TOKEN_REQUEST={$__tokenVal}";
+    xhr.onreadystatechange = function() {
```

Comment:
I think you already have on the page included jquery, so why not use `$.ajax` instead that is easier to read?